src/circuit/gadgets/pointer.rs

```diff
@@ -33,6 +33,12 @@ pub struct AllocatedPtr<F: PrimeField> {
     hash: AllocatedNum<F>,
 }
 
+impl<F: LurkField> From<AllocatedPtr<F>> for AllocatedContPtr<F> {
```

Comment:
Nit:
```
diff --git a/src/circuit/gadgets/pointer.rs b/src/circuit/gadgets/pointer.rs
index 616d2b652..45cd1101b 100644
--- a/src/circuit/gadgets/pointer.rs
+++ b/src/circuit/gadgets/pointer.rs
@@ -35,7 +35,7 @@ pub struct AllocatedPtr<F: PrimeField> {

 impl<F: LurkField> From<AllocatedPtr<F>> for AllocatedContPtr<F> {
     fn from(other: AllocatedPtr<F>) -> Self {
-        Self::from_parts(&other.tag, &other.hash)
+        AllocatedContPtr { tag: other.tag, hash: other.hash }
     }
 }

@@ -556,7 +556,7 @@ pub struct AllocatedContPtr<F: LurkField> {

 impl<F: LurkField> From<AllocatedContPtr<F>> for AllocatedPtr<F> {
     fn from(other: AllocatedContPtr<F>) -> Self {
-        Self::from_parts(other.tag, other.hash)
+        AllocatedPtr { tag: other.tag, hash: other.hash }
     }
 }
```
Also, since I have you here, that should give you the occasion to delete the unused 
```
    pub fn from_parts(tag: &AllocatedNum<F>, hash: &AllocatedNum<F>) -> Self
```
In this same file (this method does not respect [C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control))